pkg/storage/vault.go

```diff
@@ -0,0 +1,231 @@
+	}
+
+	names := []string{}
+	for _, s := range secret.Data {
+		for _, v := range s.([]interface{}) {
+			names = append(names, v.(string))
+		}
+	}
+	return names, nil
+}
+
+// Delete will delete a secret from Vault
+func (v *VaultStore) Delete(path string) error {
+	_, err := v.Client.Logical().Delete(path)
+	if err != nil {
+		return fmt.Errorf("unable to delete secret %s: %+v", path, err)
+	}
+	return nil
+}
+
+// GeneratePoliciesAndRoles will generate a set of policies for a given directory of members
+func (v *VaultStore) GeneratePoliciesAndRoles(directoryBackend, roleDir, policyDir, defaultTeam string, members []directory.Member) error {
+	policies, ok := policies[directoryBackend]
+	if !ok {
+		return fmt.Errorf("unknown directory backend %s", directoryBackend)
+	}
+
+	buf := bytes.NewBuffer([]byte{})
+
+	gt := template.Must(template.New("generalPolicy").Parse(policies.GeneralPolicyTemplate))
+	s := space{Path: keyPrefix}
+	if err := gt.Execute(buf, s); err != nil {
+		return fmt.Errorf("unable to execute template for general policy: %+v", err)
+	}
+	p := path.Join(policyDir, fmt.Sprintf("%s.hcl", filePrefix))
+	if err := ioutil.WriteFile(p, buf.Bytes(), filePerms); err != nil {
+		return fmt.Errorf("unable to write general psst policy file: %+v", err)
+	}
+
+	// Adds default role for the "all" team in GH
+	teamRoles := roleDir
+	if path.Base(roleDir) == "users" {
+		teamRoles = path.Join(path.Dir(roleDir), "teams")
+	}
+	if err := checkRole(defaultTeam, filePrefix, teamRoles); err != nil {
+		return fmt.Errorf("unable to write all team role: %v", err)
+	}
+
+	t := template.Must(template.New("policy").Parse(policies.PolicyTemplate))
+
+	for _, m := range members {
+		buf.Reset()
+		s = space{Path: getSecretPathPrefix(m.Login)}
+		if err := t.Execute(buf, s); err != nil {
+			return fmt.Errorf("unable to execute template for user %s: %+v", m.Login, err)
+		}
+		roleName := fmt.Sprintf("%s-%s", filePrefix, m.Login)
+		p = path.Join(policyDir, fmt.Sprintf("%s.hcl", roleName))
+		if err := ioutil.WriteFile(p, buf.Bytes(), filePerms); err != nil {
+			return fmt.Errorf("unable to write policy file for user %s: %+v", m.Login, err)
+		}
+
+		if err := checkRole(m.Login, roleName, roleDir); err != nil {
+			return fmt.Errorf("Unable to setup role for %s: %v", m.Login, err)
+		}
+	}
+	return nil
+}
+
+// checkRole will append the user's psst policy if it is missing. It will also add a file for new users added to the GitHub
+// organization since the last update.
+func checkRole(login, roleName, roleDir string) error {
+	filename := path.Join(roleDir, fmt.Sprintf("%s.json", login))
+	policy := ghUserPolicy{Value: ""}
+
+	if _, err := os.Stat(filename); err == nil {
+		b, err := ioutil.ReadFile(filename)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("unable to read %s", filename))
+		}
+		if err := json.Unmarshal(b, &policy); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("unable to unmarshal %s", filename))
+		}
+	}
```

Comment:
if err != nil, do you still want to continue?

Reply:
I don't necessarily want to continue because I might remove some improperly added additions for the given user. I'd rather have the user take a look and either fix the issue or delete the file.

Reply:
Then you should add an else clause and error out here.

Reply:
Sorry. If `err != nil` then we will write a general policy file down which is the work outside of this outer if block. The rest of the code here is the else clause if there is an error with `os.Stat` meaning that the file doesn't currently exist.

Reply:
If you're assuming that `err != nil` means the file doesn't exist, you should verify the error is os.IsNotExist.

Reply:
Added.